### PR TITLE
chore: deprecate enums

### PR DESCRIPTION
**Pull request by talkor**
https://monday.monday.com/boards/3532714909/pulses/7021052307
https://monday.monday.com/boards/3532714909/pulses/7021049810
https://monday.monday.com/boards/3532714909/pulses/7111673033

### CODE REVIEW

File: packages/core/src/components/AlertBanner/AlertBannerText/AlertBannerText.tsx [hunk 1]
```diff
@@ -23,12 +20,7 @@ const AlertBannerText: FC<AlertBannerTextProps> = ({ text, marginLeft = false, i
   const isOverflowing = useIsOverflowing({ ref: componentRef });
 
   return (
-    <Tooltip
-      position="bottom"
-      content={isOverflowing && text}
-      showTrigger={TOOLTIP_SHOW_TRIGGERS}
-      hideTrigger={TOOLTIP_HIDE_TRIGGERS}
-    >
+    <Tooltip position="bottom" content={isOverflowing && text} showTrigger="mouseenter" hideTrigger="mouseleave">
```
Comment (rivka-ungar) [inline packages/core/src/components/AlertBanner/AlertBannerText/AlertBannerText.tsx:23]: Is it intentional that `showTrigger` and `hideTrigger` were changed from array to string?

File: packages/core/src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.tsx [hunk 2]
```diff
@@ -76,8 +73,8 @@ const AvatarGroupCounterTooltipContainer: React.FC<AvatarGroupCounterTooltipCont
     <Tooltip
       open={isKeyboardTooltipVisible}
       hideDelay={AVATAR_GROUP_COUNTER_TOOLTIP_SHOW_DELAY}
-      showTrigger={SHOW_TRIGGER}
-      hideTrigger={HIDE_TRIGGER}
+      showTrigger="mouseenter"
+      hideTrigger="mouseleave"
```
Comment (rivka-ungar) [inline packages/core/src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.tsx:77]: Also here?

File: packages/core/src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -162,9 +163,9 @@ export interface DialogState {
 export type DialogEvent = React.MouseEvent | React.KeyboardEvent | KeyboardEvent | React.FocusEvent | CustomEvent;
 
 export default class Dialog extends PureComponent<DialogProps, DialogState> {
-  static hideShowTriggers = HideShowEvent;
-  static positions = DialogPosition;
-  static animationTypes = AnimationType;
+  static hideShowTriggers = DialogTriggerEventEnum;
+  static positions = DialogPositionEnum;
+  static animationTypes = AnimationTypeEnum;
```
Comment (rivka-ungar) [inline packages/core/src/components/Dialog/Dialog.tsx:168]: `showTrigger` and `hideTrigger` props should set default the string value?

File: packages/core/src/components/SplitButton/SplitButton.tsx [hunk 4]
```diff
@@ -10,15 +10,13 @@ import {
   DEFAULT_DIALOG_HIDE_TRIGGER,
   DEFAULT_DIALOG_SHOW_TRIGGER,
   DIALOG_MOVE_BY,
-  EMPTY_ARR,
```
Comment (rivka-ungar) [inline packages/core/src/components/SplitButton/SplitButton.tsx:13]: Is it still in use or it can be removed from the constants file?

**Review by rivka-ungar — APPROVED** [review]

File: packages/core/src/components/AlertBanner/AlertBannerText/AlertBannerText.tsx (hunk 1 above)
Comment (talkor) [reply packages/core/src/components/AlertBanner/AlertBannerText/AlertBannerText.tsx:23]: No lol

File: packages/core/src/components/SplitButton/SplitButton.tsx (hunk 4 above)
Comment (talkor) [reply packages/core/src/components/SplitButton/SplitButton.tsx:13]: Yes it's used in some other places but we will remove it in upcoming prs

**Pull request merged**